pkg/server/server.go

```diff
@@ -140,7 +140,11 @@ func (s *Server) Run(ctx context.Context) error {
 		klog.Infof("Finished bootstrapping system CRDs")
 
 		if err := wait.PollInfiniteWithContext(goContext(ctx), time.Second, func(ctx context.Context) (bool, error) {
-			if err := configshard.Bootstrap(ctx, s.KcpClusterClient.Cluster(configshard.SystemShardCluster)); err != nil {
+			if err := configshard.Bootstrap(ctx,
+				s.ApiExtensionsClusterClient.Cluster(configshard.SystemShardCluster).Discovery(),
```

Comment:
Does it matter from which clientset we grab the discovery client? I remember at some point the context/round-tripper stuff was different between the two?

Reply:
yes, it does matter, since the discovery differs across logical clusters. In this case, it is okay because `system:shard` cluster knows about the built-in types like namespaces.